broid-wechat/src/core/adapter.ts

```diff
@@ -0,0 +1,154 @@
+import * as Promise from "bluebird";
+import broidSchemas, { ISendParameters } from "broid-schemas";
+import { Logger } from "broid-utils";
+import * as fs from "fs-extra";
+import * as uuid from "node-uuid";
+import * as path from "path";
+import * as R from "ramda";
+import * as request from "request";
+import { Observable } from "rxjs/Rx";
+import * as tmp from "tmp";
+import * as WeChat from "wechat-api";
+
+import { IAdapterHTTPOptions, IAdapterOptions } from "./interfaces";
+import Parser from "./parser";
+import WebHookServer from "./webHookServer";
+
+const HTTPOptions: IAdapterHTTPOptions = {
+  host: "127.0.0.1",
+  port: 8080,
+};
+
+export default class Adapter {
+  public serviceID: string;
+
+  private appID: string;
+  private appSecret: string;
+  private client: any;
+  private connected: boolean;
+  private HTTPOptions: IAdapterHTTPOptions;
+  private logLevel: string;
+  private logger: Logger;
+  private parser: Parser;
+  private webhookServer: WebHookServer;
+
+  constructor(obj: IAdapterOptions) {
+    this.serviceID = obj && obj.serviceID || uuid.v4();
+    this.logLevel = obj && obj.logLevel || "info";
+    this.appID = obj && obj.appID;
+    this.appSecret = obj && obj.appSecret;
+
+    this.HTTPOptions = obj.http || HTTPOptions;
+
+    this.logger = new Logger("adapter", this.logLevel);
+
+    if (!this.appID) {
+      throw new Error("appID must be set");
+    }
+    if (!this.appSecret) {
+      throw new Error("appSecret must be set");
+    }
+
+    this.client = Promise.promisifyAll(new WeChat(this.appID, this.appSecret));
+    this.client.setEndpoint("api.wechat.com");
```

Comment:
may be we can put the endpoint editable via the options and keep `api.wechat.com` as default?